File: iteration2/Grade.java



// Grade class
class Grade {
    private final int NUMERICAL_GRADE;

    // Contructor
    public Grade(int numericalGrade) {
        this.NUMERICAL_GRADE = numericalGrade;
    }

    public boolean isRetakableGrade() {
        return NUMERICAL_GRADE < 50;
    }

    public boolean isPassedGrade() {
        return NUMERICAL_GRADE >= 35;
    }

    // Convert numerical grade to letter grade
    public String getGradeLetter() {
        if (NUMERICAL_GRADE >= 90 && NUMERICAL_GRADE <= 100) {
            return "AA";
        } else if (NUMERICAL_GRADE >= 80 && NUMERICAL_GRADE < 90) {
            return "BA";
        } else if (NUMERICAL_GRADE >= 70 && NUMERICAL_GRADE < 80) {
            return "BB";
        } else if (NUMERICAL_GRADE >= 60 && NUMERICAL_GRADE < 70) {
            return "CB";
        } else if (NUMERICAL_GRADE >= 50 && NUMERICAL_GRADE < 60) {
            return "CC";
        } else if (NUMERICAL_GRADE >= 40 && NUMERICAL_GRADE < 50) {
            return "DC";
        } else if (NUMERICAL_GRADE >= 35 && NUMERICAL_GRADE < 40) {
            return "DD";
        } else if (NUMERICAL_GRADE < 35 && NUMERICAL_GRADE >= 0) {
            return "FF";
        }

        return "Invalid Grade";
    }

    @Override
    public String toString() {
        return "Grade : " + NUMERICAL_GRADE + ", Letter Grade : " + getGradeLetter();
    }

    public int getNumericalGrade() {
        return NUMERICAL_GRADE;
    }
}

File: iteration3/Grade.py
class Grade:
    __numerical_grade = 0

    def __init__(self, numerical_grade):
        self.__numerical_grade = numerical_grade

    def is_retakable_grade(self):
        return self.__numerical_grade < 50

    def is_passed_grade(self):
        return self.__numerical_grade < 35

    def get_letter_grade(self):
        if 90 <= self.__numerical_grade <= 100:
            return "AA"
        elif 80 <= self.__numerical_grade < 90:
            return "BA"
        elif 70 <= self.__numerical_grade < 80:
            return "BB"
        elif 60 <= self.__numerical_grade < 70:
            return "CB"
        elif 50 <= self.__numerical_grade < 60:
            return "CC"
        elif 40 <= self.__numerical_grade < 50:
            return "DC"
        elif 35 <= self.__numerical_grade < 40:
            return "DD"
        elif 35 > self.__numerical_grade >= 0:
            return "FF"
        return "Invalid Grade"

    def __str__(self):
        return f"Grade: {self.__numerical_grade} {", Letter Grade : "} {self.get_letter_grade()}"

    def get_numerical_grade(self):
        return self.__numerical_grade
